csrc/ir_base_nodes.cpp

```diff
@@ -112,6 +112,15 @@ const std::vector<Expr*>& Val::uses() const {
   return uses_;
 }
 
+bool Val::addUse(Expr* expr) {
+  if (std::find(uses_.begin(), uses_.end(), expr) == uses_.end()) {
+    uses_.push_back(expr);
+    fusion()->invalidateUses();
```

Comment:
Do we need to invalidate every time we add uses?

For scalars, that shouldn't be necessary

Reply:
You're right. I'll add a check here